cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/PeriodicReplicationReceiver.java

```diff
@@ -0,0 +1,46 @@
+package com.cloudant.sync.replication;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.content.WakefulBroadcastReceiver;
+
+/**
+ * This class implements a {@link WakefulBroadcastReceiver} that handles events related to
+ * periodic replication.  It responds to reboot of the device to trigger the resetting of
+ * the {@link android.app.AlarmManager} used to trigger periodic replication, and it
+ * handles the periodic alarms sent by the {@link android.app.AlarmManager} to trigger
+ * the periodic replications.
+ *
+ * The resetting of periodic alarms after reboot and the handling of the periodic alarms
+ * are delegated to the {@link PeriodicReplicationService} associated with this
+ * {@link android.content.BroadcastReceiver}.
+ *
+ * @param <T> The {@link PeriodicReplicationService} component triggered by this
+ * {@link android.content.BroadcastReceiver}
+ */
+public class PeriodicReplicationReceiver<T extends PeriodicReplicationService> extends WakefulBroadcastReceiver {
+
+    static final String ALARM_ACTION = "com.cloudant.sync.replication.PeriodicReplicationReceiver.Alarm";
+
+    Class<T> clazz;
+
+    protected PeriodicReplicationReceiver(Class<T> clazz) {
+        this.clazz = clazz;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        int command = PeriodicReplicationService.COMMAND_NONE;
+        if (ALARM_ACTION.equals(intent.getAction())) {
```

Comment:
I think it's worth it. Although I don't think having a null `Intent` makes much sense I can't see there's any guarantee that it can't happen, so probably best to be defensive. Fixed in c12068f.